interfaces/mount/backend.go

```diff
@@ -86,28 +86,40 @@ func (b *Backend) Remove(snapName string) error {
 
 // deriveContent computes .fstab tables based on requests made to the specification.
 func deriveContent(spec *Specification, snapInfo *snap.Info) map[string]*osutil.FileState {
-	// No entries? Nothing to do!
-	if len(spec.mountEntries) == 0 {
-		return nil
-	}
-	// Compute the contents of the fstab file. It should contain all the mount
-	// rules collected by the backend controller.
-	var buffer bytes.Buffer
-	for _, entry := range spec.mountEntries {
-		fmt.Fprintf(&buffer, "%s\n", entry)
-	}
-	fstate := &osutil.FileState{Content: buffer.Bytes(), Mode: 0644}
 	content := make(map[string]*osutil.FileState)
-	// Add the new per-snap fstab file. This file will be read by snap-confine.
-	content[fmt.Sprintf("snap.%s.fstab", snapInfo.Name())] = fstate
-	// Add legacy per-app/per-hook fstab files. Those are identical but
-	// snap-confine doesn't yet load it from a per-snap location. This can be
-	// safely removed once snap-confine is updated.
-	for _, appInfo := range snapInfo.Apps {
-		content[fmt.Sprintf("%s.fstab", appInfo.SecurityTag())] = fstate
+	if len(spec.mountEntries) != 0 {
+		// Compute the contents of the fstab file. It should
+		// contain all the mount rules collected by the
+		// backend controller.
+		var buffer bytes.Buffer
+		for _, entry := range spec.mountEntries {
+			fmt.Fprintf(&buffer, "%s\n", entry)
+		}
+		fstate := &osutil.FileState{Content: buffer.Bytes(), Mode: 0644}
+		// Add the new per-snap fstab file. This file will be
+		// read by snap-confine.
+		content[fmt.Sprintf("snap.%s.fstab", snapInfo.Name())] = fstate
+		// Add legacy per-app/per-hook fstab files. Those are
```

Comment:
I think we can drop those now, nobody reads them.

Reply:
Okay.  I'll add that change to the branch.